e2e-tests/src/merod.rs

```diff
@@ -55,34 +55,84 @@ impl Merod {
                     server_host,
                     "--swarm-port",
-                    swarm_port.to_string().as_str(),
+                    &swarm_port.to_string(),
                     "--server-port",
-                    server_port.to_string().as_str(),
+                    &server_port.to_string(),
                 ],
                 "init",
             )
             .await?;
+
         let result = child.wait().await?;
         if !result.success() {
             bail!("Failed to initialize node '{}'", self.name);
         }
 
-        let config_args = ["config"].into_iter().chain(args);
+        // Clone for safe reuse
+        let config_args: Vec<String> = args.iter().map(|s| s.to_string()).collect();
 
-        let mut child = self.run_cmd(config_args, "config").await?;
-        let result = child.wait().await?;
-        if !result.success() {
-            bail!("Failed to configure node '{}'", self.name);
+        let config_changed = !config_args.is_empty();
+
+        if config_changed {
+            // Reuse original args for chaining
+            let mut child = self
+                .run_cmd(["config", "set"].into_iter().chain(args.iter().copied()), "config")
+                .await?;
+
+            let result = child.wait().await?;
+            if !result.success() {
+                bail!("Failed to configure node '{}'", self.name);
+            }
+        } else {
+            // Check for output format
+            let output_format = args
+                .iter()
+                .find(|arg| arg.starts_with("--output-format"))
+                .and_then(|arg| arg.split('=').nth(1));
+
+            let print_format = match output_format {
+                Some("json") => "--format json",
+                _ => "",
+            };
+
+            let mut child = self
+                .run_cmd(
+                    ["config", "print"]
+                        .into_iter()
+                        .chain(if print_format.is_empty() {
+                            None
+                        } else {
+                            Some(print_format)
+                        }),
+                    "config-print",
+                )
+                .await?;
+
+            let result = child.wait().await?;
+            if !result.success() {
+                bail!("Failed to print node '{}' configuration", self.name);
+            }
         }
 
         Ok(())
     }
 
+    pub async fn hints(&self) -> EyreResult<()> {
+        let hints = r#"
+        Hints:
+        - sync.timeout_ms: Valid values are any positive integer in milliseconds.
+        - sync.interval_ms: Valid values are any positive integer in milliseconds.
+        - network.swarm.port: The port for the network swarm, valid range is 1024-65535.
+        - network.server.listen: A list of addresses the server will listen on (e.g., "127.0.0.1:8080").
+        "#;
+
+        self.output_writer.write_str(hints);
+        Ok(())
+    }
```

Comment:
these should not be static, if you cannot figure out a path to dynamically reflect and fetch the fields, variants and description of config entries, we can reduce the scope to what you have so far